hecate/src/main.rs

```diff
@@ -134,29 +136,49 @@ fn database_check(conn_str: &String, is_read: bool) {
             if !is_read {
                 match conn.query("
                     SELECT
-                        (regexp_matches(version(), 'PostgreSQL (.*?) '))[1]::FLOAT AS postgres_v,
-                        (regexp_matches(postgis_version(), '^(.*?) '))[1]::FLOAT AS postgis_v
+                        (regexp_matches(version(), 'PostgreSQL (.*?) '))[1] AS postgres_v,
+                        (regexp_matches(postgis_version(), '^(.*?) '))[1] AS postgis_v
                 ", &[]) {
                     Ok(res) => {
                         if res.len() != 1 {
-                            println!("ERROR: Connection unable obtain postgres version using ({}) {}", conn_type, conn_str);
+                            println!("ERROR: Connection unable obtain PostgreSQL version using ({}) {}", conn_type, conn_str);
                             std::process::exit(1);
                         }
-
-                        let postgres_v: f64 = res.get(0).get(0);
-                        if postgres_v < hecate::POSTGRES {
-                            println!("ERROR: Hecate requires a min postgres version of {}", hecate::POSTGRES);
-                            std::process::exit(1);
+                        let postgres_v: String = res.get(0).get(0);
+                        let want_postgres_v = VersionReq::parse(hecate::POSTGRES_VERSION).unwrap();
```

Comment:
Let's add a `match` statement here with a human readable error message if it fails. (Obviously it shouldn't - but I fat finger things all the time)